File: frameworks/framework/lua-bindings/api/NetProtocol.lua

--------------------------------
-- @module NetProtocol
-- @extend Ref

--------------------------------
-- @function [parent=#NetProtocol] setID 
-- @param self
-- @param #string str
        
--------------------------------
-- @function [parent=#NetProtocol] getID 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- @function [parent=#NetProtocol] init 
-- @param self
-- @return bool#bool ret (return value: bool)
        
--------------------------------
-- @function [parent=#NetProtocol] setServiceID 
-- @param self
-- @param #string str
        
--------------------------------
-- @function [parent=#NetProtocol] getServiceID 
-- @param self
-- @return string#string ret (return value: string)
        
--------------------------------
-- @function [parent=#NetProtocol] getHandler 
-- @param self
-- @return NetHandler#NetHandler ret (return value: NetHandler)
        
--------------------------------
-- @function [parent=#NetProtocol] setHandler 
-- @param self
-- @param #NetHandler nethandler
        
--------------------------------
-- overload function: create(string, string, NetHandler)
--          
-- overload function: create()
--          
-- @function [parent=#NetProtocol] create
-- @param self
-- @param #string str
-- @param #string str
-- @param #NetHandler nethandler
-- @return NetProtocol#NetProtocol ret (retunr value: NetProtocol)

return nil

File: frameworks/CocosLua/net/NetProtocol.cpp
//
//  NetProtocol.cpp
//  Framework
//
//  Created by wp_g4 on 13-3-2.
//
//

#include "NetProtocol.h"

NetProtocol::NetProtocol()
	: handler(NULL)
{
}

bool NetProtocol::init()
{
    return true;
}

NetProtocol::~NetProtocol()
{
    CC_SAFE_RELEASE_NULL(handler);
}

NetProtocol* NetProtocol::create(const std::string &protocolID, const std::string &serviceID, NetHandler *handler)
{
    NetProtocol* netProtocol=NetProtocol::create();
    netProtocol->setID(protocolID);
    netProtocol->setServiceID(serviceID);
    netProtocol->setHandler(handler);
    return netProtocol;
}